tests/chart/test_dag_deploy_networkpolicy.py

```diff
@@ -30,6 +30,34 @@ def test_gsr_networkpolicy_defaults_with_authsidecar_enabled(self, kube_version)
         )
         assert len(docs) == 0
 
+    def test_dag_deploy_networkpolicy_dag_deploy_enabled_with_dataplane_mode(self, kube_version):
+        """Test that a valid networkPolicy are rendered when dag-deploy is enabled."""
```

Comment:
We should parametrize this test for all three modes. Also need to fix the docstring:
```suggestion
    def test_dag_deploy_networkpolicy_dag_deploy_enabled_with_dataplane_mode(self, kube_version):
        """Test that a valid networkPolicy is rendered when dag-deploy and networkPolicies are enabled."""
```